tutorial.md

```diff
@@ -0,0 +1,212 @@
+# Google Cloud Client Libraries for Rust
+
+This guide explains how to use the Google Cloud Client Libraries for Rust to interact with Google Cloud services in your applications.
+
+> **Note**: These libraries are currently in development and not recommended for production use.
+
+## Before you begin
+
+1. [Install Rust](https://www.rust-lang.org/tools/install)
+2. [Set up a Google Cloud project](https://cloud.google.com/resource-manager/docs/creating-managing-projects)
+3. [Install and initialize the Google Cloud CLI](https://cloud.google.com/sdk/docs/install)
+
+## Installing the client libraries
+
+Add the required dependencies to your `Cargo.toml`:
+
+```toml
+[dependencies]
+google-cloud-secretmanager-v1 = { version = "0.1", features = ["unstable-stream"] }
+tokio = { version = "1.0", features = ["macros", "rt-multi-thread"] }
+```
+
+## Authenticating
+
+### Using Application Default Credentials (Recommended)
+
+```rust
+use google_cloud_secretmanager_v1::client::SecretManagerServiceClient;
+
+async fn init_client() -> Result<SecretManagerServiceClient, Box<dyn std::error::Error>> {
+    // Creates a client using Application Default Credentials
+    let client = SecretManagerServiceClient::new().await?;
+    Ok(client)
+}
+```
+
+### Using Service Account Keys
+
+```rust
+use google_cloud_auth::credentials::CredentialsFile;
+use google_cloud_secretmanager_v1::client::SecretManagerServiceClient;
+
+async fn init_client_with_credentials() -> Result<SecretManagerServiceClient, Box<dyn std::error::Error>> {
+    // Load credentials from a service account key file
+    let credentials = CredentialsFile::new_from_file(
+        "/path/to/service-account-key.json"
+    ).await?;
+    
+    // Initialize client with explicit credentials
+    let client = SecretManagerServiceClient::builder()
+        .with_credentials(credentials)
+        .build()
+        .await?;
+    
+    Ok(client)
+}
+```
+
+## Using the client libraries
+
+### Basic operations
+
+#### Listing secrets
+
+```rust
+use google_cloud_secretmanager_v1::client::SecretManagerServiceClient;
```

Comment:
I think the client name is simply `SecretManagerService`.  This is why we write the code samples as actual code in `guide/samples/*` and embed the code using `{{#include }}` to ensure the code actually compiles.

Please refactor all the code compilable code samples.